components/script/dom/window.rs

```diff
@@ -893,7 +894,14 @@ impl WindowMethods for Window {
     /// <https://html.spec.whatwg.org/multipage/#dom-window-requestanimationframe>
     fn RequestAnimationFrame(&self, callback: Rc<FrameRequestCallback>) -> u32 {
         self.Document()
-            .request_animation_frame(AnimationFrameCallback::FrameRequestCallback { callback })
+            .request_animation_frame(AnimationFrameCallback::FrameRequestCallback {
+                callback,
+                webgl_chan: if cfg!(target_os = "macos") && opts::get().with_io_surface {
```

Comment:
We should do this unconditionally so it's easier to reason about the implementation. We can make the swap operation in the backend a no-op if necessary.